doc/source/tune/user-guide.rst

```diff
@@ -295,49 +295,157 @@ When running a hyperparameter search, Tune can automatically and periodically sa
 
 .. code-block:: python
 
-    import os
-    import time
+    import ray
     from ray import tune
+    from your_module import my_trainable
+
+    ray.init(address="<cluster-IP>:<port>")  # set `address=None` to train on laptop
+
+    # configure how checkpoints are sync'd to the scheduler/sampler 
+    # we recommend cloud storage checkpointing as it survives the cluster when
+    # instances are terminated, and has better performance
+    sync_config = tune.syncConfig(
+        upload_dir="s3://my-checkpoints-bucket/path/",  # requires AWS credentials
+        sync_to_cloud=True,
+        sync_to_driver=False  # recommended for performance when using cloud checkpointing 
+    )
 
-    def train_func(config, checkpoint_dir=None):
-        start = 0
-        if checkpoint_dir:
-            with open(os.path.join(checkpoint_dir, "checkpoint")) as f:
-                state = json.loads(f.read())
-                start = state["step"] + 1
+    # here we wrap our Ray model (or custom training func) with a couple of things:
+    # 1) with_parameters() ensures when we checkpoint, weights are saved (not just hyperparameters)
+    # 2) durable() ensures that this trainer can be saved to cloud storage (ie: S3)
+    durable_trainer_with_params = tune.with_parameters(tune.durable(my_trainable))
 
-        for step in range(start, 100):
-            time.sleep(1)
+    # this starts the run!
+    tune.run(
+        durable_trainer_with_params,
+
+        # name of your experiment 
+        name="my-tune-exp",
+
+        # a directory where results are stored before being
+        # sync'd to head node/cloud storage
+        local_dir="/tmp/mypath",
 
-            # Obtain a checkpoint directory
-            with tune.checkpoint_dir(step=step) as checkpoint_dir:
-                path = os.path.join(checkpoint_dir, "checkpoint")
-                with open(path, "w") as f:
-                    f.write(json.dumps({"step": step}))
+        # see above! we will sync our checkpoints to S3 directory
+        sync_config=sync_config,
 
-            tune.report(hello="world", ray="tune")
+        # we'll checkpoint every 10 iterations, keep the best five
+        # checkpoints (by AUC score, descending), and always checkpoint
+        # at the end of the last run!
+        checkpoint_score_attr="max-auc",
+        keep_checkpoints_num=5,
+        checkpoint_freq=10,
+        checkpoint_at_end=True,
+
+        # a very useful trick! this will resume from the last run specified by
+        # sync_config (if one exists), otherwise it will start a new tuning run
+        resume="AUTO",
+    )
 
-    tune.run(train_func)
+In this example, checkpoints will be saved:
 
-In this example, checkpoints will be saved by training iteration to ``local_dir/exp_name/trial_name/checkpoint_<step>``.
+* **Locally**: not saved! we set ``sync_to_driver=False``, so nothing will be sync'd to the driver (your laptop)
```

Comment:
If you are using Ray Client, *nothing* will be synced to the laptop, no matter the sync config. `driver` in sync_config, confusingly, is the machine where the Tune driver process is running - in case of Ray Client, that will be the cluster head node, not laptop

Reply:
oh that is super confusing. so it really should be named `sync_to_head`, yes?

Reply:
It made more sense before we had ray client, but yeah, it is confusing now

Reply:
@worldveil This still needs addressing

Reply:
@Yard1 can you adjust the wording here in a suggestion (CMD+G) ? haven't gotten the chance to fully understand here and might just be quicker to have to change the wording

Reply:
```suggestion
* **Locally**: not saved! Nothing will be sync'd to the driver (your laptop) automatically
```